### PR TITLE
doc: update PR workflow for GH merge queues

### PR DESCRIPTION
**Pull request by Pixep**

### CODE REVIEW

File: docs/project/pull_request_workflow.md [hunk 1]
```diff
@@ -101,29 +102,28 @@ review you typically want to leave the development history undisturbed until the
 end so that comments on any particular increment aren't lost. We typically use
 the GitHub squash-and-merge functionality to land things.
 
-### Managing pull requests with multiple commits
```
Comment (Pixep) [inline docs/project/pull_request_workflow.md:104]: @chandlerc (being the author of that part) These are still good general recommendations but don't really apply anymore. Are you ok with removing this section, or do you prefer to keep it?

Comment (jonmeow) [reply docs/project/pull_request_workflow.md:104]: I'd be fine removing this, I don't think we've ever done what's described here -- we pretty exclusively squash-and-merge, not rebase.

**Review by jonmeow — APPROVED** [review]
Looks good to me, but might want to give another day or two for @chandlerc to comment if he wants to.

**Comment by jonmeow** [comment]
NB: This was stuck on an action config issue, I pushed an empty commit to unjam it.

**Pull request merged**